MinPlatformPkg/Test/Library/TestPointCheckLib/StandaloneMmTestPointCheckLib.c

```diff
@@ -0,0 +1,429 @@
+        NULL,
+        TEST_POINT_INDEX_BYTE6_SMM,
+        TEST_POINT_BYTE6_SMM_READY_TO_LOCK_SECURE_SMM_COMMUNICATION_BUFFER
+        );
+    } else {
+      TestPointLibClearFeaturesVerified (
+        PLATFORM_TEST_POINT_ROLE_PLATFORM_IBV,
+        NULL,
+        TEST_POINT_INDEX_BYTE6_SMM,
+        TEST_POINT_BYTE6_SMM_READY_TO_LOCK_SECURE_SMM_COMMUNICATION_BUFFER
+        );
+    }
+  }
+
+Done:
+  FreePool (CommData);
+
+  DEBUG ((DEBUG_INFO, "======== TestPointStandaloneMmReadyToBootStandaloneMmPageProtectionHandler - Exit\n"));
+  return EFI_SUCCESS;
+}
+
+/**
+  Dispatch function for a Software SMI handler.
+
+  Caution: This function may receive untrusted input.
+  Communicate buffer and buffer size are external input, so this function will do basic validation.
+
+  @param DispatchHandle  The unique handle assigned to this handler by SmiHandlerRegister().
+  @param Context         Points to an optional handler context which was specified when the
+                         handler was registered.
+  @param CommBuffer      A pointer to a collection of data in memory that will
+                         be conveyed from a non-Standalone MM environment into an Standalone MM environment.
+  @param CommBufferSize  The size of the CommBuffer.
+
+  @retval EFI_SUCCESS Command is handled successfully.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmHandler (
+  IN EFI_HANDLE  DispatchHandle,
+  IN CONST VOID  *Context         OPTIONAL,
+  IN OUT VOID    *CommBuffer      OPTIONAL,
+  IN OUT UINTN   *CommBufferSize  OPTIONAL
+  )
+{
+  TEST_POINT_SMM_COMMUNICATION_HEADER  CommData;
+  UINTN                                TempCommBufferSize;
+
+  //
+  // If input is invalid, stop processing this SMI
+  //
+  if ((CommBuffer == NULL) || (CommBufferSize == NULL)) {
+    return EFI_SUCCESS;
+  }
+
+  TempCommBufferSize = *CommBufferSize;
+
+  if (TempCommBufferSize < sizeof (TEST_POINT_SMM_COMMUNICATION_HEADER)) {
+    DEBUG ((DEBUG_ERROR, "TestPointStandaloneMmHandler: MM communication buffer size invalid!\n"));
+    return EFI_SUCCESS;
+  }
+
+  CopyMem (&CommData, CommBuffer, sizeof (CommData));
+  if (CommData.Version != TEST_POINT_SMM_COMMUNICATION_VERSION) {
+    DEBUG ((DEBUG_ERROR, "TestPointStandaloneMmHandler: MM communication Version invalid!\n"));
+    return EFI_SUCCESS;
+  }
+
+  switch (CommData.FuncId) {
+    case TEST_POINT_SMM_COMMUNICATION_FUNC_ID_UEFI_GCD_MAP_INFO:
+      return TestPointStandaloneMmReadyToBootStandaloneMmPageProtectionHandler (CommBuffer, CommBufferSize);
+  }
+
+  return EFI_SUCCESS;
+}
+
+/**
+  This service verifies the system state within SMM after Exit Boot Services is invoked.
+
+  @retval EFI_SUCCESS         The test point check was performed successfully.
+**/
+EFI_STATUS
+EFIAPI
+TestPointStandaloneMmExitBootServices (
+  VOID
+  )
+{
+  return TestPointMmExitBootServices ();
+}
+
+/**
+  Register Mm Test Point handler.
+**/
+VOID
+RegisterStandaloneMmTestPointHandler (
```

Comment:
I think I saw `RegisterSmmTestPointHandler` in the traditional MM instance as well. Just wondering if there is any particular reason why we leave them as separate implementations?

Reply:
Consolidated.